<a href="https://colab.research.google.com/github/vibeeshK/Olympic-medals-and-elevation-levels/blob/main/provect_v2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Part 1: Hypothesis Testing: Olympic Medal Count and Elevation by Country** 

*Vibeesh Kamalakannan, Rasa Blourtchi, Raghav Arora*

## **Introduction**

"I'm a mountain guy" - Khabib Nurmagomedov. Khabib Nurmagomedov, the longest reigning UFC lightweight champion credits his physicality and athleticism to his environment, saying that living and training in the mountains, was one of the reasons for his success. This sparked the start our analysis, to see whether the environmental factor of altitute really has an impact on an athletes success. And what better way to test this theory, than on the biggest sporting event in the world... The Olympics! 

"https://www.youtube.com/watch?v=U4PW800YC-A"

## **Variables**

MedalCount: Represents the number of medals (bronze, silver and gold) won by a Country from 1896 to 2016

Average Elevation: The mean elevation (altitude) of a country 

Highest Point: The max elevation (altitude) of a country

Average_Population: Population averaged from 1960 to 2021

Average_GDP: GDP averaged from 1960 to 2021



## **Hypothesis**

As the Average Elevation or Highest Point of a country increases, the country's respective medal count in the olympics also increases. 

This is because training in higher altitudes the body triggers a hormonal response that enhances the way oxygen is delivered and utilised throughout the body. Creating more blood vessels for oxygen to flow through, altitude training may lead to improved heart functionality, enhanced muscle performance and greater overall health. With more oxygen flowing through your body, recovery times are minimised while strength and endurance is enhanced. Morevoer, studies have also suggested a positive impact on stress reduction and sleep patterns, making altitude training a workout style that offers ongoing benefits.

"https://www.surgefitness.com.au/blog/what-are-the-benefits-of-altitude-training/#:~:text=Creating%20more%20blood%20vessels%20for,strength%20and%20endurance%20is%20enhanced"


## **Libraries Used**

In [ ]:
import pandas as pd 
import json
from urllib.request import urlopen # library to read data from a URL
import matplotlib.pyplot as plt # visualization library we will use that integrates with pandas
import seaborn as sns
from bokeh.plotting import figure, show
from bokeh.models import ColumnDataSource, HoverTool
from bokeh.layouts import gridplot
import bokeh
import requests 
from bs4 import BeautifulSoup
import re 
from matplotlib.pyplot import figure
import numpy as np
from scipy.stats.stats import pearsonr
import plotly.express as px
import pandas as pd
import urllib.request
import zipfile
from pandas._libs.algos import diff_2d

plt.style.use('dark_background')
figure(figsize=(8, 6), dpi=80)

<ipython-input-121-e04393522cf1>:15: DeprecationWarning:

Please use `pearsonr` from the `scipy.stats` namespace, the `scipy.stats.stats` namespace is deprecated.



<Figure size 640x480 with 0 Axes>

<Figure size 640x480 with 0 Axes>

## **Data Import and Pre-Processing**

### **Highest Point**

In [ ]:
# List of highest elevation
url = "https://flagpedia.net/lists/highest-point"
data = requests.get(url).text

soup = BeautifulSoup(data, 'html.parser')
tables = soup.find_all('td')

# Create function to remove html from the extracted strings
CLEANR = re.compile('<.*?>') 
def cleanhtml(raw_html):
  cleantext = re.sub(CLEANR, '', raw_html)
  return cleantext

cleanhtml(str(tables[1]))
clean_values = []
for i in tables:
  clean_values.append(cleanhtml(str(i)))
clean_values[0:20]

ele=''
clean_values=[i for i in clean_values if i!=ele]
clean_values[0:20]

df = [] 
rows = [] 
for i in clean_values:
  if len(rows)==4:
    df.append(rows)
    rows = []
    rows.append(i)
  else:
    rows.append(i)
    
highest_points = pd.DataFrame(df, columns = ['Rank', 'Country','Highest Point','Height'])  
highest_points.head()  
     

Rank   Country     Highest Point   Height
0   1.     China     Mount Everest  8 848 m
1   2.     Nepal     Mount Everest  8 848 m
2   3.  Pakistan                K2  8 611 m
3   4.     India     Kangchenjunga  8 586 m
4   5.    Bhutan  Gangkhar Puensum  7 570 m

### **Average Elevation**

In [ ]:
# List of avg elevation
url = "https://www.atlasbig.com/en-us/countries-average-elevation"


data = requests.get(url).text

soup = BeautifulSoup(data, 'html.parser')
tables = soup.find_all('td')


cleanhtml(str(tables[1]))
clean_values = []
for i in tables:
  clean_values.append(cleanhtml(str(i)))
clean_values[0:20]

ele=''
clean_values=[i for i in clean_values if i!=ele]
clean_values[0:20]



df = [] 
rows = [] 
for i in clean_values:
  if len(rows)==2:
    df.append(rows)
    rows = []
    rows.append(i)
  else:
    rows.append(i)

average_elevation = pd.DataFrame(df, columns = ['Country','Average Elevation'])  
average_elevation.head()

Country Average Elevation
0      Bhutan             3,280
1       Nepal             3,265
2  Tajikistan             3,186
3  Kyrgyzstan             2,989
4     Lesotho             2,161

### **Olympic Data (1986-2016)**

In [ ]:
# The URL of the zip file containing the CSV file
url = 'https://github.com/vibeeshK/olmpic-dataset/blob/main/Olympic_Athletes%20(1).zip?raw=true'

# The name of the CSV file within the zip file
files_ =   ['athlete_events.csv','athlete_events_data_dictionary.csv', 'country_definitions.csv','country_definitions_data_dictionary.csv']
#csv_filename

# Download the zip file
zip_filename, _ = urllib.request.urlretrieve(url)

# Extract the CSV file from the zip file
with zipfile.ZipFile(zip_filename) as zip_file:
    with zip_file.open('athlete_events.csv') as csv_file:
        # Read the CSV file into a pandas DataFrame
        athlete_events = pd.read_csv(csv_file)

# Extract the CSV file from the zip file
with zipfile.ZipFile(zip_filename) as zip_file:
    with zip_file.open('athlete_events_data_dictionary.csv') as csv_file:
        # Read the CSV file into a pandas DataFrame
        athlete_events_data_dictionary = pd.read_csv(csv_file)

# Extract the CSV file from the zip file
with zipfile.ZipFile(zip_filename) as zip_file:
    with zip_file.open('country_definitions.csv') as csv_file:
        # Read the CSV file into a pandas DataFrame
        country_definitions = pd.read_csv(csv_file)

# Extract the CSV file from the zip file
with zipfile.ZipFile(zip_filename) as zip_file:
    with zip_file.open('country_definitions_data_dictionary.csv') as csv_file:
        # Read the CSV file into a pandas DataFrame
        country_definitions_data_dictionary = pd.read_csv(csv_file)
athlete_events.head()

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season       City          Sport  \
0  CHN  1992 Summer  1992  Summer  Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer     London           Judo   
2  DEN  1920 Summer  1920  Summer  Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer      Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter    Calgary  Speed Skating   

                              Event Medal  
0       Basketball Men's Basketball   NaN  
1      Judo Men's Extra-Lightweight   NaN  
2           Football Men's Football   NaN  
3       Tug-Of-War Men's Tug-Of-War  Gold  
4  Speed Skating Women's 500 metres   NaN

## **Data Cleaning and Merging** 

### **Olympic Data**

In [ ]:
# Split catagorical data
athlete_events['IsBronze'] = (athlete_events['Medal'] == 'Bronze').astype(int)
athlete_events['IsSilver'] = (athlete_events['Medal'] == 'Silver').astype(int)
athlete_events['IsGold'] = (athlete_events['Medal'] == 'Gold').astype(int)

In [ ]:
# Merge to get regions and NOC
olympic = pd.merge(athlete_events, country_definitions[['NOC', 'region']], on='NOC', how='left') 
olympic2 = olympic 

### **Highest Point**

In [ ]:
# Clean the highest point data
for i in range(len(olympic['region'])):
    # replace hardik with shardul
    if olympic['region'][i] == 'Boliva':
        olympic['region'][i] = 'Bolivia'

    elif olympic['region'][i] == 'Czech Republic':
        olympic['region'][i] = 'Czechia'

    elif olympic['region'][i] == 'Democratic Republic of the Congo':
        olympic['region'][i] = 'DR Congo'

    elif olympic['region'][i] == 'Ivory Coast':
        olympic['region'][i] = "Côte d'Ivoire (Ivory Coast)"

    elif olympic['region'][i] == 'Macedonia':
        olympic['region'][i] = "North Macedonia"

    elif olympic['region'][i] == 'Ivory Coast':
        olympic['region'][i] = "Côte d'Ivoire (Ivory Coast)"
    
    elif olympic['region'][i] == 'Republic of Congo':
        olympic['region'][i] = "Republic of the Congo"
      
    elif olympic['region'][i] == 'Saint Kitts':
        olympic['region'][i] = "Saint Kitts and Nevis"

    elif olympic['region'][i] == 'Saint Vincent':
        olympic['region'][i] = "Saint Vincent and the Grenadines"

    elif olympic['region'][i] == 'Swaziland':
        olympic['region'][i] = "Eswatini (Swaziland)"

    elif olympic['region'][i] == 'Trinidad':
        olympic['region'][i] = "Trinidad and Tobago"

    elif olympic['region'][i] == 'UK':
        olympic['region'][i] = "United Kingdom"

    elif olympic['region'][i] == 'USA':
        olympic['region'][i] = "United States"

    elif olympic['region'][i] == 'Virgin Islands, British':
        olympic['region'][i] = "British Virgin Islands"
    
    elif olympic['region'][i] == 'Virgin Islands, US':
        olympic['region'][i] = "United States Virgin Islands"
      
olympic.head()

<ipython-input-127-fcbff94120e0>:41: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

<ipython-input-127-fcbff94120e0>:23: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

<ipython-input-127-fcbff94120e0>:47: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

<ipython-input-127-fcbff94120e0>:14: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season       City          Sport  \
0  CHN  1992 Summer  1992  Summer  Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer     London           Judo   
2  DEN  1920 Summer  1920  Summer  Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer      Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter    Calgary  Speed Skating   

                              Event Medal  IsBronze  IsSilver  IsGold  \
0       Basketball Men's Basketball   NaN         0         0       0   
1      Judo Men's Extra-Lightweight   NaN         0         0       0   
2           Football Men's Football   NaN         0         0       0   
3       Tug-Of-War Men's Tug-Of-War  Gold         0         0       1   
4  Speed Skating Women's 500 metres   NaN         0         0       0   

        region  
0        China  
1        China  
2      Denmark  
3      Denmark  
4  Netherlands

In [ ]:
# Group Medal Count by Country and Merge with Highest Elevation
clean_olympic = olympic
clean_olympic = clean_olympic.groupby('region')['IsGold', 'IsSilver', 'IsBronze'].sum()
clean_olympic['MedalCount'] = (clean_olympic['IsBronze']+clean_olympic['IsSilver']+clean_olympic['IsGold']).astype(int)
clean_olympic = clean_olympic.reset_index()
clean_olympic = clean_olympic.rename(columns={'region': 'Country'})
clean_olympic_highest = pd.merge(clean_olympic, highest_points[['Country', 'Height']], on='Country', how='left')
clean_olympic_highest = clean_olympic_highest.rename(columns={'Height': 'Highest Point'}) 
clean_olympic_highest = clean_olympic_highest[clean_olympic_highest['Highest Point'].notna()]
clean_olympic_highest['Highest Point'] = clean_olympic_highest['Highest Point'].str.replace(',', '') 
clean_olympic_highest['Highest Point'] = clean_olympic_highest['Highest Point'].str.replace('m', '') 
clean_olympic_highest['Highest Point'] = clean_olympic_highest['Highest Point'].str.replace(r'\s+', '', regex=True).astype(int)
clean_olympic_highest['Highest Point'] = pd.to_numeric(clean_olympic_highest['Highest Point']) 
clean_olympic_highest.head()

<ipython-input-128-6ae70b3a7c43>:3: FutureWarning:

Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.



Country  IsGold  IsSilver  IsBronze  MedalCount  Highest Point
0     Afghanistan       0         0         2           2           7492
1         Albania       0         0         0           0           2764
2         Algeria       5         4         8          17           3003
3  American Samoa       0         0         0           0            966
4         Andorra       0         0         0           0           2942

### **Average Elevation**

In [ ]:
# Clean the avg elevation data
for i in range(len(olympic2['region'])):
    # replace hardik with shardul
    if olympic2['region'][i] == 'Boliva':
        olympic2['region'][i] = 'Bolivia'

    elif olympic2['region'][i] == 'Democratic Republic of the Congo':
        olympic2['region'][i] = 'Congo-Kinshasa'

    elif olympic2['region'][i] == 'Dominica':
        olympic2['region'][i] = 'Dominican Republic'

    elif olympic2['region'][i] == 'Ivory Coast':
        olympic2['region'][i] = "Côte d'Ivoire" 	

    elif olympic2['region'][i] == 'Netherlands':
        olympic2['region'][i] = "The Netherlands"

    elif olympic2['region'][i] == 'Republic of Congo':
        olympic2['region'][i] = "Congo-Brazzaville"
    
    elif olympic2['region'][i] == 'Russia':
        olympic2['region'][i] = "Russian Federation"

    elif olympic2['region'][i] == 'South Sudan':
        olympic2['region'][i] = "Sudan"

    elif olympic2['region'][i] == 'Trinidad':
        olympic2['region'][i] = "Trinidad and Tobago"

    elif olympic2['region'][i] == 'UK':
        olympic2['region'][i] = "United Kingdom"

    elif olympic2['region'][i] == 'United States':
        olympic2['region'][i] = "United States of America"

olympic2.head()


<ipython-input-129-df301051336e>:17: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

<ipython-input-129-df301051336e>:35: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

<ipython-input-129-df301051336e>:23: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

<ipython-input-129-df301051336e>:11: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season       City          Sport  \
0  CHN  1992 Summer  1992  Summer  Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer     London           Judo   
2  DEN  1920 Summer  1920  Summer  Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer      Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter    Calgary  Speed Skating   

                              Event Medal  IsBronze  IsSilver  IsGold  \
0       Basketball Men's Basketball   NaN         0         0       0   
1      Judo Men's Extra-Lightweight   NaN         0         0       0   
2           Football Men's Football   NaN         0         0       0   
3       Tug-Of-War Men's Tug-Of-War  Gold         0         0       1   
4  Speed Skating Women's 500 metres   NaN         0         0       0   

            region  
0            China  
1            China  
2          Denmark  
3          Denmark  
4  The Netherlands

In [ ]:
# Group Medal Count by Country and Merge with Average Elevation
clean_olympic2 = olympic2
clean_olympic2 = clean_olympic2.groupby('region')['IsGold', 'IsSilver', 'IsBronze'].sum()
clean_olympic2['MedalCount'] = (clean_olympic2['IsBronze']+clean_olympic2['IsSilver']+clean_olympic2['IsGold']).astype(int)
clean_olympic2 = clean_olympic2.reset_index()
clean_olympic2 = clean_olympic2.rename(columns={'region': 'Country'})
clean_olympic_avg = pd.merge(clean_olympic2, average_elevation[['Country', 'Average Elevation']], on='Country', how='left')
clean_olympic_avg = clean_olympic_avg[clean_olympic_avg['Average Elevation'].notna()]
clean_olympic_avg['Average Elevation'] = clean_olympic_avg['Average Elevation'].str.replace(',', '') 
clean_olympic_avg['Average Elevation'] = pd.to_numeric(clean_olympic_avg['Average Elevation'])   
clean_olympic_avg.head()

<ipython-input-132-41c83791d2af>:3: FutureWarning:

Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.



Country  IsGold  IsSilver  IsBronze  MedalCount  Average Elevation
0  Afghanistan       0         0         2           2               1884
1      Albania       0         0         0           0                708
2      Algeria       5         4         8          17                800
4      Andorra       0         0         0           0               1996
5       Angola       0         0         0           0               1112

## **Data Visualization** 

### **Highest Point**

In [ ]:
# Highest point per country vs medals won

fig = px.scatter(clean_olympic_highest, x='Highest Point', y='MedalCount', hover_name='Country',
                 category_orders={"Highest Point": clean_olympic_highest["Highest Point"].sort_values().unique()}) 
fig.show()

### **Average Elevation**

In [ ]:
# Avg elevation vs medals

fig = px.scatter(clean_olympic_avg, x='Average Elevation', y='MedalCount', hover_name='Country',
                 category_orders={"Average Elevation": clean_olympic_avg["Average Elevation"].sort_values().unique()}) 
fig.show()

## **T Test: Pearson Coefficient** 

### **Highest Point**

In [ ]:
# Pearson coefficient to find correlation between highest point and medal

pearsonr(clean_olympic_highest["Highest Point"], clean_olympic_highest["MedalCount"])


PearsonRResult(statistic=0.17584309656065222, pvalue=0.012526842829910926)

Pearson correlation coefficient (r): 0.1758 /
Two-tailed p-value: 0.01252

Since the correlation coefficient is close to 1, this tells us that there is a no association between the highest point and medals won by the country. 

### **Average Elevation**

In [ ]:
# Pearson coefficient to find correlation between avg height and medal

pearsonr(clean_olympic_avg["Average Elevation"], clean_olympic_avg["MedalCount"]) 


PearsonRResult(statistic=-0.0889243353378809, pvalue=0.2649979729790312)

Pearson correlation coefficient (r):-0.08892 / 
Two-tailed p-value: 0.264997

Since the correlation coefficient is close to 1, this tells us that there is a no association between the average elevation and medals won by the country. 



## **Conclusion**

The reason we believe our analysis failed to show a relationship between Medal Count and Elevation was because we performed this analysis on a global level. By doing so,  we were unable to keep other important factors such as GDP, Education, etc constant which are known to have a stronger impact on Olympic Medal count.


## **Next Steps**



*   Conduct Analysis only on Countries with a similar socio-economic background.
*   Conduct Analysis on Summer vs. Winter Olympics and Sport type.







# **Part 2: Hypothesis Testing: Olympic Medal Count and Elevation by Country**
### *Summer Olympics vs. Winter Olympics vs. Sport Type*

## **Data Processing and Filtering**

In [ ]:
olympic2

ID                      Name Sex   Age  Height  Weight  \
0            1                 A Dijiang   M  24.0   180.0    80.0   
1            2                  A Lamusi   M  23.0   170.0    60.0   
2            3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3            4      Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4            5  Christine Jacoba Aaftink   F  21.0   185.0    82.0   
...        ...                       ...  ..   ...     ...     ...   
271111  135569                Andrzej ya   M  29.0   179.0    89.0   
271112  135570                  Piotr ya   M  27.0   176.0    59.0   
271113  135570                  Piotr ya   M  27.0   176.0    59.0   
271114  135571        Tomasz Ireneusz ya   M  30.0   185.0    96.0   
271115  135571        Tomasz Ireneusz ya   M  34.0   185.0    96.0   

                  Team  NOC        Games  Year  Season            City  \
0                China  CHN  1992 Summer  1992  Summer       Barcelona   
1                China  CHN  2012 Summer  2012  Summer          London   
2              Denmark  DEN  1920 Summer  1920  Summer       Antwerpen   
3       Denmark/Sweden  DEN  1900 Summer  1900  Summer           Paris   
4          Netherlands  NED  1988 Winter  1988  Winter         Calgary   
...                ...  ...          ...   ...     ...             ...   
271111        Poland-1  POL  1976 Winter  1976  Winter       Innsbruck   
271112          Poland  POL  2014 Winter  2014  Winter           Sochi   
271113          Poland  POL  2014 Winter  2014  Winter           Sochi   
271114          Poland  POL  1998 Winter  1998  Winter          Nagano   
271115          Poland  POL  2002 Winter  2002  Winter  Salt Lake City   

                Sport                                     Event Medal  \
0          Basketball               Basketball Men's Basketball   NaN   
1                Judo              Judo Men's Extra-Lightweight   NaN   
2            Football                   Football Men's Football   NaN   
3          Tug-Of-War               Tug-Of-War Men's Tug-Of-War  Gold   
4       Speed Skating          Speed Skating Women's 500 metres   NaN   
...               ...                                       ...   ...   
271111           Luge                Luge Mixed (Men)'s Doubles   NaN   
271112    Ski Jumping  Ski Jumping Men's Large Hill, Individual   NaN   
271113    Ski Jumping        Ski Jumping Men's Large Hill, Team   NaN   
271114      Bobsleigh                      Bobsleigh Men's Four   NaN   
271115      Bobsleigh                      Bobsleigh Men's Four   NaN   

        IsBronze  IsSilver  IsGold           region  
0              0         0       0            China  
1              0         0       0            China  
2              0         0       0          Denmark  
3              0         0       1          Denmark  
4              0         0       0  The Netherlands  
...          ...       ...     ...              ...  
271111         0         0       0           Poland  
271112         0         0       0           Poland  
271113         0         0       0           Poland  
271114         0         0       0           Poland  
271115         0         0       0           Poland  

[271116 rows x 19 columns]

### **Summer Olympics**

In [ ]:
summer = olympic2.loc[olympic2['Season'] == 'Summer'] 
summer = summer.groupby('region')['IsGold', 'IsSilver', 'IsBronze'].sum()
summer['MedalCount'] = (summer['IsBronze']+summer['IsSilver']+summer['IsGold']).astype(int)
summer = summer.reset_index()
summer = summer.rename(columns={'region': 'Country'})
summer_avg = pd.merge(summer, average_elevation[['Country', 'Average Elevation']], on='Country', how='left')
summer_avg = summer_avg[summer_avg['Average Elevation'].notna()]
summer_avg['Average Elevation'] = summer_avg['Average Elevation'].str.replace(',', '') 
summer_avg['Average Elevation'] = pd.to_numeric(summer_avg['Average Elevation'])   
summer_avg

<ipython-input-138-a077454b4cb2>:2: FutureWarning:

Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.



Country  IsGold  IsSilver  IsBronze  MedalCount  Average Elevation
0    Afghanistan       0         0         2           2               1884
1        Albania       0         0         0           0                708
2        Algeria       5         4         8          17                800
4        Andorra       0         0         0           0               1996
5         Angola       0         0         0           0               1112
..           ...     ...       ...       ...         ...                ...
198    Venezuela       2         3        10          15                450
199      Vietnam       1         3         0           4                398
200        Yemen       0         0         0           0                999
201       Zambia       0         1         1           2               1138
202     Zimbabwe      17         4         1          22                961

[159 rows x 6 columns]

### **Winter Olympics**

In [ ]:
winter = olympic2.loc[olympic2['Season'] == 'Winter'] 
winter = winter.groupby('region')['IsGold', 'IsSilver', 'IsBronze'].sum()
winter['MedalCount'] = (winter['IsBronze']+winter['IsSilver']+winter['IsGold']).astype(int)
winter = winter.reset_index()
winter = winter.rename(columns={'region': 'Country'})
winter_avg = pd.merge(winter, average_elevation[['Country', 'Average Elevation']], on='Country', how='left')
winter_avg = winter_avg[winter_avg['Average Elevation'].notna()]
winter_avg['Average Elevation'] = winter_avg['Average Elevation'].str.replace(',', '') 
winter_avg['Average Elevation'] = pd.to_numeric(winter_avg['Average Elevation'])   
winter_avg

<ipython-input-139-f813fdc3f132>:2: FutureWarning:

Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.



Country  IsGold  IsSilver  IsBronze  MedalCount  \
0                     Albania       0         0         0           0   
1                     Algeria       0         0         0           0   
3                     Andorra       0         0         0           0   
4                   Argentina       0         0         0           0   
5                     Armenia       0         0         0           0   
..                        ...     ...       ...       ...         ...   
106  United States of America     166       308       161         635   
107                   Uruguay       0         0         0           0   
108                Uzbekistan       1         0         0           1   
109                 Venezuela       0         0         0           0   
110                  Zimbabwe       0         0         0           0   

     Average Elevation  
0                  708  
1                  800  
3                 1996  
4                  595  
5                 1792  
..                 ...  
106                760  
107                109  
108                450  
109                450  
110                961  

[92 rows x 6 columns]

### **Sport Type**

In [ ]:
pears = []
p_val = []
sport = []
for i in olympic2['Sport'].unique():
  try:
    sports = olympic2.loc[olympic2['Sport'] == i] 
    sports = sports.groupby('region')['IsGold', 'IsSilver', 'IsBronze'].sum()
    sports['MedalCount'] = (sports['IsBronze']+sports['IsSilver']+sports['IsGold']).astype(int)
    sports = sports.reset_index()
    sports = sports.rename(columns={'region': 'Country'})
    sports_avg = pd.merge(sports, average_elevation[['Country', 'Average Elevation']], on='Country', how='left')
    sports_avg = sports_avg[sports_avg['Average Elevation'].notna()]
    sports_avg['Average Elevation'] = sports_avg['Average Elevation'].str.replace(',', '') 
    sports_avg['Average Elevation'] = pd.to_numeric(sports_avg['Average Elevation'])   
    pears.append(pearsonr(sports_avg["Average Elevation"], sports_avg["MedalCount"])[0])
    p_val.append(pearsonr(sports_avg["Average Elevation"], sports_avg["MedalCount"])[1])
    sport.append(i)
  except:
    pass

<ipython-input-140-7cd1fff37dec>:7: FutureWarning:

Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.



## **Data Visualization**

### **Summer Olympics**

In [ ]:
fig = px.scatter(summer_avg, x='Average Elevation', y='MedalCount', hover_name='Country',
                 category_orders={"Average Elevation": summer_avg["Average Elevation"].sort_values().unique()}) 
fig.show()

As we can see, there is no clear correlation here between Medal Count and Average Elevation when looking at Summer Olympic Events.

### **Winter Olympics**

In [ ]:
fig = px.scatter(winter_avg, x='Average Elevation', y='MedalCount', hover_name='Country',
                 category_orders={"Average Elevation": winter_avg["Average Elevation"].sort_values().unique()}) 
fig.show()

As we can see, there is no clear correlation here between Medal Count and Average Elevation when looking at Winter Olympic Events.

### **Sport Type**

After looking at the Pearson Coefficients and p-values. We chose to create scatterplots for the only four Olympic Events that showed any form correlation and a significant p-value. The four sports were Trampoling, Table Tennis, Badminton, and Synchronized Swimming.

### *Trampolining*

In [ ]:
trampolining = olympic2.loc[olympic2['Sport'] == 'Trampolining'] 
trampolining = trampolining.groupby('region')['IsGold', 'IsSilver', 'IsBronze'].sum()
trampolining['MedalCount'] = (trampolining['IsBronze']+trampolining['IsSilver']+trampolining['IsGold']).astype(int)
trampolining = trampolining.reset_index()
trampolining = trampolining.rename(columns={'region': 'Country'})
trampolining_avg = pd.merge(trampolining, average_elevation[['Country', 'Average Elevation']], on='Country', how='left')
trampolining_avg = trampolining_avg[trampolining_avg['Average Elevation'].notna()]
trampolining_avg['Average Elevation'] = trampolining_avg['Average Elevation'].str.replace(',', '') 
trampolining_avg['Average Elevation'] = pd.to_numeric(trampolining_avg['Average Elevation'])   
fig = px.scatter(trampolining_avg, x='Average Elevation', y='MedalCount', hover_name='Country',
                 category_orders={"Average Elevation": trampolining_avg["Average Elevation"].sort_values().unique()}) 
fig.show()

<ipython-input-143-ac829a59e2d8>:2: FutureWarning:

Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.



### *Table Tennis*

In [ ]:
tt = olympic2.loc[olympic2['Sport'] == 'Table Tennis'] 
tt = tt.groupby('region')['IsGold', 'IsSilver', 'IsBronze'].sum()
tt['MedalCount'] = (tt['IsBronze']+tt['IsSilver']+tt['IsGold']).astype(int)
tt = tt.reset_index()
tt = tt.rename(columns={'region': 'Country'})
tt_avg = pd.merge(tt, average_elevation[['Country', 'Average Elevation']], on='Country', how='left')
tt_avg = tt_avg[tt_avg['Average Elevation'].notna()]
tt_avg['Average Elevation'] = tt_avg['Average Elevation'].str.replace(',', '') 
tt_avg['Average Elevation'] = pd.to_numeric(tt_avg['Average Elevation'])   
fig = px.scatter(tt_avg, x='Average Elevation', y='MedalCount', hover_name='Country',
                 category_orders={"Average Elevation": tt_avg["Average Elevation"].sort_values().unique()}) 
fig.show()

<ipython-input-144-36516c912ecb>:2: FutureWarning:

Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.



### *Badminton*

In [ ]:
badminton = olympic2.loc[olympic2['Sport'] == 'Trampolining'] 
badminton = badminton.groupby('region')['IsGold', 'IsSilver', 'IsBronze'].sum()
badminton['MedalCount'] = (badminton['IsBronze']+badminton['IsSilver']+badminton['IsGold']).astype(int)
badminton = badminton.reset_index()
badminton = badminton.rename(columns={'region': 'Country'})
badminton_avg = pd.merge(badminton, average_elevation[['Country', 'Average Elevation']], on='Country', how='left')
badminton_avg = badminton_avg[badminton_avg['Average Elevation'].notna()]
badminton_avg['Average Elevation'] = badminton_avg['Average Elevation'].str.replace(',', '') 
badminton_avg['Average Elevation'] = pd.to_numeric(badminton_avg['Average Elevation'])   
fig = px.scatter(badminton_avg, x='Average Elevation', y='MedalCount', hover_name='Country',
                 category_orders={"Average Elevation": badminton_avg["Average Elevation"].sort_values().unique()}) 
fig.show()

<ipython-input-145-3934dbf644d9>:2: FutureWarning:

Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.



### *Synchronized Swimming*

In [ ]:
ss = olympic2.loc[olympic2['Sport'] == 'Synchronized Swimming'] 
ss = ss.groupby('region')['IsGold', 'IsSilver', 'IsBronze'].sum()
ss['MedalCount'] = (ss['IsBronze']+ss['IsSilver']+ss['IsGold']).astype(int)
ss = ss.reset_index()
ss = ss.rename(columns={'region': 'Country'})
ss_avg = pd.merge(ss, average_elevation[['Country', 'Average Elevation']], on='Country', how='left')
ss_avg = ss_avg[ss_avg['Average Elevation'].notna()]
ss_avg['Average Elevation'] = ss_avg['Average Elevation'].str.replace(',', '') 
ss_avg['Average Elevation'] = pd.to_numeric(ss_avg['Average Elevation'])   
fig = px.scatter(ss_avg, x='Average Elevation', y='MedalCount', hover_name='Country',
                 category_orders={"Average Elevation": ss_avg["Average Elevation"].sort_values().unique()}) 
fig.show()

<ipython-input-146-72422b4d65f3>:2: FutureWarning:

Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.



## **T Test: Pearson Coefficient** 

### **Summer Olympics**

In [ ]:
pearsonr(summer_avg["Average Elevation"], summer_avg["MedalCount"])

PearsonRResult(statistic=-0.09251879284951471, pvalue=0.24608856831626558)

Pearson correlation coefficient (r):-0.0925 / 
Two-tailed p-value: 0.24608

Since the correlation coefficient is close to 0.1, this tells us that there is a no association between the average elevation and medals won by the country when considering Summer Olympic Events.

### **Winter Olmpics**

In [ ]:
pearsonr(winter_avg["Average Elevation"], winter_avg["MedalCount"])

PearsonRResult(statistic=-0.09173302121507936, pvalue=0.3844765795722149)

Pearson correlation coefficient (r):-0.0917 / Two-tailed p-value: 0.38448

Since the correlation coefficient is close to 0.1, this tells us that there is a no association between the average elevation and medals won by the country when considering Winter Olympic Events.

### **Sport Type**

In [ ]:
events = {'Sport': sport, "Pearson's Coefficient": pears,"P Value": p_val}
events = pd.DataFrame(data=events)
events = events.sort_values("Pearson's Coefficient") 
events = events.iloc[3:]
events

Sport  Pearson's Coefficient   P Value
60                   Alpinism              -0.468407  0.348775
3                  Tug-Of-War              -0.445934  0.126683
41                   Baseball              -0.278220  0.315345
20                     Hockey              -0.268303  0.113611
13           Art Competitions              -0.190063  0.200680
30                   Canoeing              -0.188705  0.098007
44               Rugby Sevens              -0.173941  0.569816
10                    Sailing              -0.168560  0.116443
28                    Cycling              -0.164781  0.081138
32          Modern Pentathlon              -0.157643  0.274234
19                 Water Polo              -0.148552  0.341753
2                    Football              -0.146523  0.191800
5        Cross Country Skiing              -0.138435  0.260238
24              Equestrianism              -0.134695  0.296592
26                     Boxing              -0.131891  0.125868
21                     Rowing              -0.112925  0.309415
11                   Biathlon              -0.111485  0.471237
23                    Fencing              -0.099226  0.381191
1                        Judo              -0.094785  0.261847
48                Ski Jumping              -0.092887  0.619191
36                    Archery              -0.092573  0.414088
4               Speed Skating              -0.075511  0.671247
25                   Shooting              -0.071333  0.442721
55                       Polo              -0.064945  0.868158
17                  Wrestling              -0.064684  0.514165
12                 Gymnastics              -0.063142  0.582869
15                   Handball              -0.059668  0.721956
6                   Athletics              -0.054056  0.499946
31                     Tennis              -0.052148  0.654597
18                       Luge              -0.048619  0.778262
8                    Swimming              -0.040443  0.626713
46           Beach Volleyball              -0.005218  0.975197
27                  Taekwondo               0.012995  0.900530
16              Weightlifting               0.016387  0.862611
53                   Skeleton               0.016596  0.935864
47                  Triathlon               0.019745  0.896374
14              Alpine Skiing               0.031080  0.781639
33             Figure Skating               0.040546  0.784379
49                    Curling               0.047984  0.859927
35                   Softball               0.057376  0.859419
0                  Basketball               0.059003  0.665775
34                       Golf               0.068033  0.676593
37                 Volleyball               0.078551  0.629952
7                  Ice Hockey               0.083043  0.674402
42        Rhythmic Gymnastics               0.083399  0.604177
51                      Rugby               0.088510  0.850328
54                   Lacrosse               0.116360  0.925755
29                     Diving               0.122411  0.323723
43           Freestyle Skiing               0.143723  0.396077
50               Snowboarding               0.200352  0.255911
40            Nordic Combined               0.201401  0.334350
22                  Bobsleigh               0.214716  0.189303
52  Short Track Speed Skating               0.243979  0.178404
9                   Badminton               0.266026  0.038242
39               Table Tennis               0.289095  0.006300
38      Synchronized Swimming               0.297408  0.073829
58        Military Ski Patrol               0.306945  0.615413
45               Trampolining               0.452422  0.026434

Above you can find all the Pearson Coefficients and p-values by Sport Type

# **Part 3: Hypothesis Testing: Olympic Medal Count and Elevation by Country**
### *Population & GDP Adjusted*

## **Data Import and Pre-Processing**

### **Population**

In [ ]:
url = 'https://github.com/vibeeshK/olmpic-dataset/raw/main/Population%20Data%20(1).xlsx'
population  = pd.read_excel(url)

population

Country Country Code     Indicator Name  \
0                          Aruba          ABW  Population, total   
1    Africa Eastern and Southern          AFE  Population, total   
2                    Afghanistan          AFG  Population, total   
3     Africa Western and Central          AFW  Population, total   
4                         Angola          AGO  Population, total   
..                           ...          ...                ...   
261                       Kosovo          XKX  Population, total   
262                  Yemen, Rep.          YEM  Population, total   
263                 South Africa          ZAF  Population, total   
264                       Zambia          ZMB  Population, total   
265                     Zimbabwe          ZWE  Population, total   

    Indicator Code         1960         1961         1962         1963  \
0      SP.POP.TOTL      54608.0      55811.0      56682.0      57475.0   
1      SP.POP.TOTL  130692579.0  134169237.0  137835590.0  141630546.0   
2      SP.POP.TOTL    8622466.0    8790140.0    8969047.0    9157465.0   
3      SP.POP.TOTL   97256290.0   99314028.0  101445032.0  103667517.0   
4      SP.POP.TOTL    5357195.0    5441333.0    5521400.0    5599827.0   
..             ...          ...          ...          ...          ...   
261    SP.POP.TOTL     947000.0     966000.0     994000.0    1022000.0   
262    SP.POP.TOTL    5542459.0    5646668.0    5753386.0    5860197.0   
263    SP.POP.TOTL   16520441.0   16989464.0   17503133.0   18042215.0   
264    SP.POP.TOTL    3119430.0    3219451.0    3323427.0    3431381.0   
265    SP.POP.TOTL    3806310.0    3925952.0    4049778.0    4177931.0   

            1964         1965  ...         2013         2014         2015  \
0        58178.0      58782.0  ...     102880.0     103594.0     104257.0   
1    145605995.0  149742351.0  ...  567891875.0  583650827.0  600008150.0   
2      9355514.0    9565147.0  ...   31541209.0   32716210.0   33753499.0   
3    105959979.0  108336203.0  ...  387204553.0  397855507.0  408690375.0   
4      5673199.0    5736582.0  ...   26147002.0   27128337.0   28127721.0   
..           ...          ...  ...          ...          ...          ...   
261    1050000.0    1078000.0  ...    1818117.0    1812771.0    1788196.0   
262    5973803.0    6097298.0  ...   26984002.0   27753304.0   28516545.0   
263   18603097.0   19187194.0  ...   53873616.0   54729551.0   55876504.0   
264    3542764.0    3658024.0  ...   15234976.0   15737793.0   16248230.0   
265    4310332.0    4447149.0  ...   13555422.0   13855753.0   14154937.0   

            2016         2017         2018         2019         2020  \
0       104874.0     105439.0     105962.0     106442.0     106585.0   
1    616377331.0  632746296.0  649756874.0  667242712.0  685112705.0   
2     34636207.0   35643418.0   36686784.0   37769499.0   38972230.0   
3    419778384.0  431138704.0  442646825.0  454306063.0  466189102.0   
4     29154746.0   30208628.0   31273533.0   32353588.0   33428486.0   
..           ...          ...          ...          ...          ...   
261    1777557.0    1791003.0    1797085.0    1788878.0    1790133.0   
262   29274002.0   30034389.0   30790513.0   31546691.0   32284046.0   
263   56422274.0   56641209.0   57339635.0   58087055.0   58801927.0   
264   16767761.0   17298054.0   17835893.0   18380477.0   18927715.0   
265   14452704.0   14751101.0   15052184.0   15354608.0   15669666.0   

            2021  Population_Average  
0       106537.0        7.698861e+04  
1    702976832.0        3.459694e+08  
2     40099462.0        1.804367e+07  
3    478185907.0        2.357148e+08  
4     34503774.0        1.451323e+07  
..           ...                 ...  
261    1786038.0        1.598995e+06  
262   32981641.0        1.554513e+07  
263   59392255.0        3.826222e+07  
264   19473125.0        8.878663e+06  
265   15993524.0        9.560026e+06  

[266 rows x 67 columns]

### **GDP**


In [ ]:
url = 'https://github.com/vibeeshK/olmpic-dataset/raw/main/GDP%20(1).csv'
gdp = pd.read_csv(url)

gdp

Country Country Code     Indicator Name  \
0                          Aruba          ABW  GDP (current US$)   
1    Africa Eastern and Southern          AFE  GDP (current US$)   
2                    Afghanistan          AFG  GDP (current US$)   
3     Africa Western and Central          AFW  GDP (current US$)   
4                         Angola          AGO  GDP (current US$)   
..                           ...          ...                ...   
261                       Kosovo          XKX  GDP (current US$)   
262                  Yemen, Rep.          YEM  GDP (current US$)   
263                 South Africa          ZAF  GDP (current US$)   
264                       Zambia          ZMB  GDP (current US$)   
265                     Zimbabwe          ZWE  GDP (current US$)   

     Indicator Code          1960          1961          1962          1963  \
0    NY.GDP.MKTP.CD           NaN           NaN           NaN           NaN   
1    NY.GDP.MKTP.CD  2.129152e+10  2.180944e+10  2.370806e+10  2.821128e+10   
2    NY.GDP.MKTP.CD  5.377778e+08  5.488889e+08  5.466667e+08  7.511112e+08   
3    NY.GDP.MKTP.CD  1.040414e+10  1.112789e+10  1.194319e+10  1.267633e+10   
4    NY.GDP.MKTP.CD           NaN           NaN           NaN           NaN   
..              ...           ...           ...           ...           ...   
261  NY.GDP.MKTP.CD           NaN           NaN           NaN           NaN   
262  NY.GDP.MKTP.CD           NaN           NaN           NaN           NaN   
263  NY.GDP.MKTP.CD  8.748597e+09  9.225996e+09  9.813996e+09  1.085420e+10   
264  NY.GDP.MKTP.CD  7.130000e+08  6.962857e+08  6.931429e+08  7.187143e+08   
265  NY.GDP.MKTP.CD  1.052990e+09  1.096647e+09  1.117602e+09  1.159512e+09   

             1964          1965  ...          2013          2014  \
0             NaN           NaN  ...  2.727933e+09  2.791061e+09   
1    2.611994e+10  2.968348e+10  ...  9.826770e+11  1.003400e+12   
2    8.000000e+08  1.006667e+09  ...  2.056449e+10  2.055058e+10   
3    1.383837e+10  1.486223e+10  ...  8.322170e+11  8.924980e+11   
4             NaN           NaN  ...  1.334020e+11  1.372440e+11   
..            ...           ...  ...           ...           ...   
261           NaN           NaN  ...  6.735731e+09  7.074658e+09   
262           NaN           NaN  ...  4.041523e+10  4.322859e+10   
263  1.195600e+10  1.306899e+10  ...  4.008860e+11  3.811990e+11   
264  8.394286e+08  1.082857e+09  ...  2.803724e+10  2.714102e+10   
265  1.217138e+09  1.311436e+09  ...  1.909102e+10  1.949552e+10   

             2015          2016          2017          2018          2019  \
0    2.963128e+09  2.983799e+09  3.092179e+09  3.202235e+09  3.368970e+09   
1    9.231440e+11  8.898590e+11  1.030480e+12  1.016700e+12  1.009050e+12   
2    1.999816e+10  1.801956e+10  1.889635e+10  1.841885e+10  1.890449e+10   
3    7.669580e+11  6.905450e+11  6.837480e+11  7.663600e+11  7.947190e+11   
4    8.721930e+10  4.984049e+10  6.897277e+10  7.779294e+10  6.930911e+10   
..            ...           ...           ...           ...           ...   
261  6.295820e+09  6.682833e+09  7.180813e+09  7.878509e+09  7.899879e+09   
262  4.244450e+10  3.131783e+10  2.684223e+10  2.160616e+10           NaN   
263  3.467100e+11  3.235860e+11  3.814490e+11  4.041600e+11  3.885320e+11   
264  2.125122e+10  2.095841e+10  2.587360e+10  2.631159e+10  2.330867e+10   
265  1.996312e+10  2.054868e+10  1.758489e+10  3.415607e+10  2.183223e+10   

             2020          2021  GDP_Average  
0    2.610039e+09  3.126019e+09   2006595301  
1    9.341790e+11  1.089450e+12  3.53538E+11  
2    2.014344e+10  1.478686e+10   7722577009  
3    7.848000e+11  8.401870e+11  2.47269E+11  
4    5.361907e+10  6.740429e+10  37939660063  
..            ...           ...          ...  
261  7.716925e+09  9.412034e+09   6780315068  
262           NaN           NaN  18230152814  
263  3.376200e+11  4.190150e+11   1.6234E+11  
264  1.811063e+10  2.214763e+10   7685751286  


## **Data Merging**

In [ ]:
# Olympic Dataset
population_olympic = pd.merge(clean_olympic_avg, population[['Country', 'Population_Average']], on='Country', how='left')
pop_gdp_olympic = pd.merge(population_olympic, gdp[['Country', 'GDP_Average']], on='Country', how='left')
pop_gdp_olympic = pop_gdp_olympic[pop_gdp_olympic['Average Elevation'].notna()]
pop_gdp_olympic['Average Elevation'] = pd.to_numeric(pop_gdp_olympic['Average Elevation'])
pop_gdp_olympic

# Summer Olympics 
population_summer = pd.merge(summer_avg, population[['Country', 'Population_Average']], on='Country', how='left')
pop_gdp_summer = pd.merge(population_summer, gdp[['Country', 'GDP_Average']], on='Country', how='left')
pop_gdp_summer = pop_gdp_summer[pop_gdp_summer['Average Elevation'].notna()] 
pop_gdp_summer['Average Elevation'] = pd.to_numeric(pop_gdp_summer['Average Elevation'])
pop_gdp_summer

# Winter Olympics 
population_winter = pd.merge(winter_avg, population[['Country', 'Population_Average']], on='Country', how='left')
pop_gdp_winter = pd.merge(population_winter, gdp[['Country', 'GDP_Average']], on='Country', how='left')
pop_gdp_winter = pop_gdp_winter[pop_gdp_winter['Average Elevation'].notna()] 
pop_gdp_winter['Average Elevation'] = pd.to_numeric(pop_gdp_winter['Average Elevation'])
pop_gdp_winter


Country  IsGold  IsSilver  IsBronze  MedalCount  \
0                    Albania       0         0         0           0   
1                    Algeria       0         0         0           0   
2                    Andorra       0         0         0           0   
3                  Argentina       0         0         0           0   
4                    Armenia       0         0         0           0   
..                       ...     ...       ...       ...         ...   
87  United States of America     166       308       161         635   
88                   Uruguay       0         0         0           0   
89                Uzbekistan       1         0         0           1   
90                 Venezuela       0         0         0           0   
91                  Zimbabwe       0         0         0           0   

    Average Elevation  Population_Average  GDP_Average  
0                 708        2.712054e+06   7188044170  
1                 800        2.560887e+07  69838425359  
2                1996        4.952982e+04   1627000982  
3                 595        3.277564e+07  2.11517E+11  
4                1792        2.932083e+06   6503475442  
..                ...                 ...          ...  
87                760        2.563010e+08  7.93559E+12  
88                109        3.087452e+06  19074008782  
89                450        2.068203e+07  35058405025  
90                450                 NaN          NaN  
91                961        9.560026e+06   8134379498  

[92 rows x 8 columns]

## **Data Binning** 

### **Olympic Dataset**

In [ ]:
cols_to_convert = ['IsGold', 'IsSilver', 'IsBronze','MedalCount', 'Average Elevation', 'Population_Average', 'GDP_Average']
pop_gdp_olympic[cols_to_convert] = pop_gdp_olympic[cols_to_convert].apply(pd.to_numeric, errors='coerce')

# create Population_bin and GDP_bin columns
pop_gdp_olympic['Population_bin'] = pd.qcut(pop_gdp_olympic['Population_Average'], q=10, labels=False)
pop_gdp_olympic['GDP_bin'] = pd.qcut(pop_gdp_olympic['GDP_Average'], q=10, labels=False)

# create Country_bin column
pop_bins = pd.qcut(pop_gdp_olympic['Population_Average'], q=10, labels=False, duplicates='drop')
gdp_bins = pd.qcut(pop_gdp_olympic['GDP_Average'], q=10, labels=False, duplicates='drop')
pop_gdp_olympic['Country_bin'] = pd.Series(pop_bins + gdp_bins, dtype=int)
pop_gdp_olympic['Country_bin'] = pd.cut(pop_gdp_olympic['Country_bin'], bins=10, labels=['<10%', '10-20%', '20-30%', '30-40%', '40-50%', '50-60%', '60-70%', '70-80%', '80-90%', '90-100%'])

# print the resulting dataframe
pop_gdp_olympic

Country  IsGold  IsSilver  IsBronze  MedalCount  Average Elevation  \
0    Afghanistan       0         0         2           2               1884   
1        Albania       0         0         0           0                708   
2        Algeria       5         4         8          17                800   
3        Andorra       0         0         0           0               1996   
4         Angola       0         0         0           0               1112   
..           ...     ...       ...       ...         ...                ...   
154    Venezuela       2         3        10          15                450   
155      Vietnam       1         3         0           4                398   
156        Yemen       0         0         0           0                999   
157       Zambia       0         1         1           2               1138   
158     Zimbabwe      17         4         1          22                961   

     Population_Average   GDP_Average  Population_bin  GDP_bin Country_bin  
0          1.804367e+07  7.722577e+09             7.0      3.0      50-60%  
1          2.712054e+06  7.188044e+09             2.0      2.0      20-30%  
2          2.560887e+07  6.983843e+10             7.0      6.0      70-80%  
3          4.952982e+04  1.627001e+09             0.0      0.0        <10%  
4          1.451323e+07  3.793966e+10             6.0      6.0      60-70%  
..                  ...           ...             ...      ...         ...  
154                 NaN           NaN             NaN      NaN         NaN  
155        6.603165e+07  1.075760e+11             9.0      7.0      80-90%  
156                 NaN           NaN             NaN      NaN         NaN  
157        8.878663e+06  7.685751e+09             5.0      3.0      40-50%  
158        9.560026e+06  8.134379e+09             5.0      3.0      40-50%  

[159 rows x 11 columns]

### **Summer Olympics**

In [ ]:
cols_to_convert = ['IsGold', 'IsSilver', 'IsBronze','MedalCount', 'Average Elevation', 'Population_Average', 'GDP_Average']
pop_gdp_summer[cols_to_convert] = pop_gdp_summer[cols_to_convert].apply(pd.to_numeric, errors='coerce')

# create Population_bin and GDP_bin columns
pop_gdp_summer['Population_bin'] = pd.qcut(pop_gdp_summer['Population_Average'], q=10, labels=False)
pop_gdp_summer['GDP_bin'] = pd.qcut(pop_gdp_summer['GDP_Average'], q=10, labels=False)

# create Country_bin column
pop_bins = pd.qcut(pop_gdp_summer['Population_Average'], q=10, labels=False, duplicates='drop')
gdp_bins = pd.qcut(pop_gdp_summer['GDP_Average'], q=10, labels=False, duplicates='drop')
pop_gdp_summer['Country_bin'] = pd.Series(pop_bins + gdp_bins, dtype=int)
pop_gdp_summer['Country_bin'] = pd.cut(pop_gdp_summer['Country_bin'], bins=10, labels=['<10%', '10-20%', '20-30%', '30-40%', '40-50%', '50-60%', '60-70%', '70-80%', '80-90%', '90-100%'])

# print the resulting dataframe
pop_gdp_summer

Country  IsGold  IsSilver  IsBronze  MedalCount  Average Elevation  \
0    Afghanistan       0         0         2           2               1884   
1        Albania       0         0         0           0                708   
2        Algeria       5         4         8          17                800   
3        Andorra       0         0         0           0               1996   
4         Angola       0         0         0           0               1112   
..           ...     ...       ...       ...         ...                ...   
154    Venezuela       2         3        10          15                450   
155      Vietnam       1         3         0           4                398   
156        Yemen       0         0         0           0                999   
157       Zambia       0         1         1           2               1138   
158     Zimbabwe      17         4         1          22                961   

     Population_Average   GDP_Average  Population_bin  GDP_bin Country_bin  
0          1.804367e+07  7.722577e+09             7.0      3.0      50-60%  
1          2.712054e+06  7.188044e+09             2.0      2.0      20-30%  
2          2.560887e+07  6.983843e+10             7.0      6.0      70-80%  
3          4.952982e+04  1.627001e+09             0.0      0.0        <10%  
4          1.451323e+07  3.793966e+10             6.0      6.0      60-70%  
..                  ...           ...             ...      ...         ...  
154                 NaN           NaN             NaN      NaN         NaN  
155        6.603165e+07  1.075760e+11             9.0      7.0      80-90%  
156                 NaN           NaN             NaN      NaN         NaN  
157        8.878663e+06  7.685751e+09             5.0      3.0      40-50%  
158        9.560026e+06  8.134379e+09             5.0      3.0      40-50%  

[159 rows x 11 columns]

### **Winter Olympics**



In [ ]:
cols_to_convert = ['IsGold', 'IsSilver', 'IsBronze','MedalCount', 'Average Elevation', 'Population_Average', 'GDP_Average']
pop_gdp_winter[cols_to_convert] = pop_gdp_winter[cols_to_convert].apply(pd.to_numeric, errors='coerce')

# create Population_bin and GDP_bin columns
pop_gdp_winter['Population_bin'] = pd.qcut(pop_gdp_winter['Population_Average'], q=10, labels=False)
pop_gdp_winter['GDP_bin'] = pd.qcut(pop_gdp_winter['GDP_Average'], q=10, labels=False)

# create Country_bin column
pop_bins = pd.qcut(pop_gdp_winter['Population_Average'], q=10, labels=False, duplicates='drop')
gdp_bins = pd.qcut(pop_gdp_winter['GDP_Average'], q=10, labels=False, duplicates='drop')
pop_gdp_winter['Country_bin'] = pd.Series(pop_bins + gdp_bins, dtype=int)
pop_gdp_winter['Country_bin'] = pd.cut(pop_gdp_winter['Country_bin'], bins=10, labels=['<10%', '10-20%', '20-30%', '30-40%', '40-50%', '50-60%', '60-70%', '70-80%', '80-90%', '90-100%'])

# print the resulting dataframe
pop_gdp_winter

Country  IsGold  IsSilver  IsBronze  MedalCount  \
0                    Albania       0         0         0           0   
1                    Algeria       0         0         0           0   
2                    Andorra       0         0         0           0   
3                  Argentina       0         0         0           0   
4                    Armenia       0         0         0           0   
..                       ...     ...       ...       ...         ...   
87  United States of America     166       308       161         635   
88                   Uruguay       0         0         0           0   
89                Uzbekistan       1         0         0           1   
90                 Venezuela       0         0         0           0   
91                  Zimbabwe       0         0         0           0   

    Average Elevation  Population_Average   GDP_Average  Population_bin  \
0                 708        2.712054e+06  7.188044e+09             1.0   
1                 800        2.560887e+07  6.983843e+10             7.0   
2                1996        4.952982e+04  1.627001e+09             0.0   
3                 595        3.277564e+07  2.115170e+11             7.0   
4                1792        2.932083e+06  6.503475e+09             1.0   
..                ...                 ...           ...             ...   
87                760        2.563010e+08  7.935590e+12             9.0   
88                109        3.087452e+06  1.907401e+10             1.0   
89                450        2.068203e+07  3.505841e+10             6.0   
90                450                 NaN           NaN             NaN   
91                961        9.560026e+06  8.134379e+09             4.0   

    GDP_bin Country_bin  
0       1.0      10-20%  
1       5.0      60-70%  
2       0.0        <10%  
3       7.0      70-80%  
4       0.0        <10%  
..      ...         ...  
87      9.0     90-100%  
88      3.0      20-30%  
89      4.0      50-60%  
90      NaN         NaN  
91      1.0      20-30%  

[92 rows x 11 columns]

## **T Test: Pearson Coefficient** 

### **Olympic Dataset**

In [ ]:
pears = []
p_val = []
bin_val = []
bins = ['<10%','10-20%','20-30%','30-40%','40-50%','50-60%','60-70%','70-80%','80-90%','90-100%']

for i in bins:

  data = pop_gdp_olympic_bin = pop_gdp_olympic.loc[pop_gdp_olympic['Country_bin'] == i]
  pears.append(pearsonr(data["Average Elevation"], data["MedalCount"])[0])
  p_val.append(pearsonr(data["Average Elevation"], data["MedalCount"])[1])
  bin_val.append(i)

gdp_all = pd.DataFrame()
gdp_all['Bin'] = bin_val
gdp_all["Pearson's Coefficient"] = pears
gdp_all['P Value'] = p_val
gdp_all

Bin  Pearson's Coefficient   P Value
0     <10%               0.033800  0.908678
1   10-20%               0.632370  0.049776
2   20-30%              -0.198360  0.461460
3   30-40%              -0.191685  0.358667
4   40-50%              -0.052648  0.852175
5   50-60%              -0.461563  0.297153
6   60-70%              -0.030742  0.910015
7   70-80%              -0.523627  0.054641
8   80-90%              -0.361651  0.203887
9  90-100%              -0.123096  0.675040

### **Summer Olympics**

In [ ]:
pears = []
p_val = []
bin_val = []
for i in bins:
  data = pop_gdp_summer.loc[pop_gdp_summer['Country_bin'] == i]
  pears.append(pearsonr(data["Average Elevation"], data["MedalCount"])[0])
  p_val.append(pearsonr(data["Average Elevation"], data["MedalCount"])[1])
  bin_val.append(i)

summer_pop_gdp = pd.DataFrame()
summer_pop_gdp['Bin'] = bin_val
summer_pop_gdp["Pearson's Coefficient"] = pears
summer_pop_gdp['P Value'] = p_val
summer_pop_gdp

Bin  Pearson's Coefficient   P Value
0     <10%               0.033800  0.908678
1   10-20%               0.632370  0.049776
2   20-30%              -0.188975  0.483342
3   30-40%              -0.173733  0.406236
4   40-50%              -0.051132  0.856392
5   50-60%              -0.493822  0.260024
6   60-70%              -0.186698  0.488724
7   70-80%              -0.565549  0.035048
8   80-90%              -0.340769  0.233156
9  90-100%              -0.119403  0.684324

## **Winter Olympics**

In [ ]:
pears = []
p_val = []
bin_val = [] 
for i in bins:
  data = pop_gdp_winter.loc[pop_gdp_winter['Country_bin'] == i]
  pears.append(pearsonr(data["Average Elevation"], data["MedalCount"])[0])
  p_val.append(pearsonr(data["Average Elevation"], data["MedalCount"])[1])
  bin_val.append(i) 

winter_pop_gdp = pd.DataFrame()
winter_pop_gdp['Bin'] = bin_val
winter_pop_gdp["Pearson's Coefficient"] = pears
winter_pop_gdp['P Value'] = p_val
winter_pop_gdp = winter_pop_gdp.iloc[1:]
winter_pop_gdp

/usr/local/lib/python3.9/dist-packages/scipy/stats/_stats_py.py:4424: ConstantInputWarning:

An input array is constant; the correlation coefficient is not defined.



Bin  Pearson's Coefficient   P Value
1   10-20%              -0.361349  0.379138
2   20-30%              -0.170229  0.616783
3   30-40%              -0.195165  0.674951
4   40-50%              -0.102149  0.809800
5   50-60%              -0.621669  0.262914
6   60-70%               0.063881  0.835754
7   70-80%              -0.586772  0.096729
8   80-90%               0.128987  0.917655
9  90-100%              -0.135868  0.690390

## **Data Visualization**

### **Olympic Dataset**

